### ISSUE TITLE
Not working with $q.resolve()

### ISSUE THREAD

**Issue by jnssnmrcs** [showpad/angular-q-spread#14]
Example that does not work:
```
$q.all([
	$q.resolve(true)
])
.spread(function(results)
{
	console.log(results)
});
```

Gives error:
`TypeError: $q.all(...).spread is not a function`

However if I do one call to `$q.defer()` before, it works:
```
$q.defer();
$q.all([
	$q.resolve(true)
])
.spread(function(results)
{
	console.log(results)
});
```

**Comment by klaascuvelier** [comment]
Thanks for reporting this @jnssnmrcs I was able to reproduce this and created a fix #16 based upon your working example.
It's a bit of a dirty fix, but it does the trick for now

**Comment by jnssnmrcs** [comment]
Sweet!
